detect_secrets/plugins/common/util.py

```diff
@@ -8,50 +8,81 @@
 from importlib import import_module
 
 from detect_secrets.plugins.base import BasePlugin
-from detect_secrets.util import get_root_directory
 
 
+def _change_custom_plugin_paths_to_tuple(custom_plugin_paths_function):
```

Comment:
I generally prefer genericized naming too, but as written the argument has to be named `custom_plugin_paths_function`, otherwise it won't work, so I'm ambivalent. Maybe I can try to make it so the decorator tuplifies all arguments 🤔 , ++.

If we put it in the function body, I don't think we could `lru_cache` it, right?

Reply:
I'm not sure if I understand why the parameter has to be named `custom_plugin_paths_function`.

I don't know where the `lru_cache` is applied right now, so we can leave it as a decorator if it works as intended already.

Reply:
*Ouch my bad @OiCMudkips, I mis-typed last Friday, the argument to the function being decorated has to be named `custom_plugin_paths`, as it's currently written. So it's not applicable to any function or any function with 1 arg, only 1 arg func's with `custom_plugin_paths` as an arg.

I will try to make it so that it can accept any 1 arg func, so we can name it a more genericized name though.

Reply:
re: `where the lru_cache is applied`
I wrap functions that accept a list, to turn them into functions that accept a tuple, before the `lru_cache` decorator gets to them, so that they can be cached. Lists are unacceptable for immutability reasons IIRC

Reply:
Following up with this: the reason the arg must be named `custom_plugin_paths`, is because any function decorated by this decorator may be called with `custom_plugin_paths` as a keyword argument. 

i.e. If I changed this to
```python
    def wrapper_of_custom_plugin_paths_function(one_arg):
        return custom_plugin_paths_function(tuple(one_arg))
```

it would result in
```
============================================== ERRORS ===============================================
________________________________ ERROR collecting tests/main_test.py ________________________________
tests/main_test.py:76: in <module>
    class TestMain:
tests/main_test.py:294: in TestMain
    'name': 'Base64HighEntropyString',
tests/main_test.py:33: in get_list_of_plugins
    for name, plugin in import_plugins(custom_plugin_paths=[]).items():
E   TypeError: wrapper_of_custom_plugin_paths_function() got an unexpected keyword argument 'custom_plugin_paths'
```